var/spack/repos/builtin/packages/clang-format/package.py

```diff
@@ -0,0 +1,245 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+import glob
+
+
+class ClangFormat(CMakePackage):
+    """clang-format formats C, C++, Obj-C, Java, JavaScript, TypeScript code"""
+
+    homepage = "https://clang.llvm.org/"
+    url      = "https::/clang.llvm.org/docs/ClangFormat.html"
+
+    # Build dependency
+    depends_on('cmake@3.4.3:', type='build')
+
+    # Universal dependency
+    depends_on('python@2.7:2.8', when='@:4.999')
+    depends_on('python')
+    depends_on('py-lit', type=('build', 'run'))
+
+    base_url = 'http://llvm.org/releases/%%(version)s/%(pkg)s-%%(version)s.src.tar.xz'
+    llvm_url = base_url % {'pkg': 'llvm'}
+
+    resources = {
+        'libcxx': {
+            'url': base_url % {'pkg': 'libcxx'},
+            'destination': 'projects',
+            'placement': 'libcxx',
+        },
+        'cfe': {
+            'url': base_url % {'pkg': 'cfe'},
+            'destination': 'tools',
+            'placement': 'clang'
+        },
+        'clang-tools-extra': {
+            'url':  base_url % {'pkg': 'clang-tools-extra'},
+            'destination': 'tools/clang/tools',
+            'placement': 'extra',
+            'variant': '+clang',
+        },
+    }
+
+    releases = [
+        {
+            'version': 'trunk',
```

Comment:
Should this be ‘develop’?

Reply:
I used `trunk` to keep the naming convention consistent with [spack's `llvm` package]( https://github.com/spack/spack/blob/133dd7a4ac7b621df9d2296ec933c2b44bcac3ed/var/spack/repos/builtin/packages/llvm/package.py#L170).

Reply:
I would prefer develop as this gives consistent version comparison

Reply:
Should I change the `llvm` package also?

Reply:
Sounds good to me

Reply:
Done in #9320.

Reply:
(for llvm, anyway)

Reply:
You can change this to develop now.